source/core/solid_base.cc

```diff
@@ -79,7 +103,68 @@ template <int dim, int spacedim>
 void
 SolidBase<dim, spacedim>::setup_triangulation(const bool restart)
 {
-  if (param->solid_mesh.type == Parameters::Mesh::Type::gmsh)
+  if (param->solid_mesh.type == Parameters::Mesh::Type::gmsh &&
```

Comment:
Would not it be clearer/faster to have cascade ifs? Like :
```
if (param->solid_mesh.type == Parameters::Mesh::Type::gmsh)
  {
  if (param->solid_mesh.simplex)
   {...}
  else
   {...}
  }
if (param->solid_mesh.type == Parameters::Mesh::Type::dealii)
  {
  if (param->solid_mesh.simplex)
   {...}
  else
   {...}
  }
```
This would have several advantages : 

- removing the `param->!solid_mesh.simplex`, as it is the logical complementary of `param->solid_mesh.simplex` (better to write "else" for that),
- in terms of speed I believe it is faster, as you have an upper layer of 2 conditions, then of 2 sub-conditions, rather than potentially 4 conditions to check (it would be on Python, but I am not fully aware of speed designs on C++ yet),
- maybe remove redundancy in the code if any, like here if you have common things to do if the mesh if of type gmsh, be it simplex of not. Or the other way around : things to do if the mesh is simplex, be it generated by gmsh of dealii (and then you will have to invert the if conditions : put the `simplex` condition in the upper layer and `mesh.type` as a sub-condition)

Reply:
I don't think speed matters here, but readability does. I like the idea of making a cascade, it increases readability. In this case Olivier, could you also refactor it in the source/core/grid so that it follows the same idea?

Reply:
@jeannejoachim I really like the idea of the cascade. I didn't think about it so that's why I didn't do it that way, but you are right regarding the readability. And Bruno, I'll do it there too.